backend/src/main/java/de/tobias/intestinalinspector/service/AppUserService.java

```diff
@@ -0,0 +1,34 @@
+package de.tobias.intestinalinspector.service;
+
+import de.tobias.intestinalinspector.model.AppUserEntity;
+import de.tobias.intestinalinspector.repository.AppUserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class AppUserService {
+
+    private final AppUserRepository appUserRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    @Autowired
+    public AppUserService(AppUserRepository appUserRepository, PasswordEncoder passwordEncoder) {
+        this.appUserRepository = appUserRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public AppUserEntity create(AppUserEntity newUser) {
+
+        newUser.setUserRole("user");
+        newUser.setUserPassword(passwordEncoder.encode(newUser.getUserPassword()));
+        AppUserEntity createdEntity = appUserRepository.save(newUser);
+        return createdEntity;
+    }
```

Comment:
It is best pratice to not modify input arguments. Think about another service also using AppUserService, you would not expect to alter `newUser` just by calling `create` on it.

I suggest creating a new user. Add `@Builder(toBuilder=true)` to `AppUserEntity` and then do something like so
```java
return appUserRepository.save(newUser.toBuilder()
    .userRole("user")
    .userPassword(passwordEncoder.encode(newUser.getUserPassword()))
    .build());